gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/organizations/environments/domains/CertificatesResource.java

```diff
@@ -257,6 +259,15 @@ private CertificateEntity filterCertificateInfos(Certificate certificate, int ce
                 filteredCertificate.setStatus(CertificateStatus.WILL_EXPIRE);
             }
         }
+        if (certificate.getConfiguration() != null) {
+            final JsonObject config = JsonObject.mapFrom(Json.decodeValue(certificate.getConfiguration(), HashMap.class));
+            if (config != null && config.getJsonArray("use") != null) {
+                filteredCertificate.setUsage(config.getJsonArray("use").stream().map(Object::toString).toList());
+            } else {
+                filteredCertificate.setUsage(List.of());
+            }
+        }
+
```

Comment:
**Suggestion/Question**: Wondering if the logic to extract the _Use_ values should be moved into the **certificate-api** module into a Configuration Utility class :thinking: